pyuvdata/utils.py

```diff
@@ -1518,27 +1529,258 @@ def uvcalibrate(
     undo : bool, optional
         If True, undo the provided calibration. i.e. apply the calibration with
         flipped gain_convention. Flag propagation rules apply the same.
+    override_ant_check : bool
+        Option to override the check that all antennas with data on the UVData
+        object are present in the UVCal object. If this option is set to True,
+        uvcalibrate will proceed without erroring and data for antennas without
+        calibrations will be flagged.
+    override_time_check : bool
+        Option to override the check that times match between the UVCal and UVData
+        objects. Only used in UVCal has only one time.
 
     Returns
     -------
     UVData, optional
         Returns if not inplace
 
     """
-    # deepcopy for not inplace
+    # copy for not inplace
     if not inplace:
-        uvdata = copy.deepcopy(uvdata)
+        uvdata = uvdata.copy()
+
+    uvdata_unique_nums = np.unique(np.append(uvdata.ant_1_array, uvdata.ant_2_array))
+    uvdata.antenna_names = np.array(uvdata.antenna_names)
```

Comment:
np.asarray could be better here, as it won't copy the data if it is already an array.